lib/commons.rb

```diff
@@ -34,7 +34,27 @@ def self.find_missing_files(commons_uploads)
   end
 
   def self.get_urls(commons_uploads, update_service: nil)
-    url_query = build_url_query commons_uploads
+    filename_query = build_info_query(commons_uploads)
+    filename_response = new(filename_query).get_image_data('pageid', '')
+    image_page_ids = []
+
+    filename_response.each do |file_data|
+      title = file_data['title']
+      page_id = file_data['pageid']
+
+      # mediawiki can generate thumbnails for jpg,png,tiff,wav files
+      # mediawiki cannot generate thumbnails for pdf,djvu files
+      if title.match?(/\.(pdf|djvu)$/i)
+        bad_file = CommonsUpload.find_by(file_name: title)
+        save_placeholder_thumbnail(bad_file) if bad_file
```

Comment:
I don't think is should be done within the `get_urls` method. The method name (`get_something`) implies that it's a read-only method will find and return data without changing the database, so this will be an unexpected side effect.

The UploadImporter#import_urls method would be a more appropriate place to handle setting a placeholder thumbnail.